lib/jdomodels/src/main/java/org/sagebionetworks/competition/dao/CompetitionDAOImpl.java

```
@@ -37,65 +47,87 @@
-			"SELECT COUNT(" + ID + ") FROM " + 
-			SQLConstants.TABLE_COMPETITION + " WHERE "+ 
-			ID + "=:" + ID;
+	private static final String SQL_ETAG_FOR_UPDATE = SQL_ETAG_WITHOUT_LOCK + " FOR UPDATE";
 	
 	private static final RowMapper<CompetitionDBO> rowMapper = ((new CompetitionDBO()).getTableMapping());
 
+	private static final String COMPETITION_NOT_FOUND = "Competition could not be found with id :";
+
 	@Override
 	@Transactional(readOnly = false, propagation = Propagation.REQUIRED)
 	public String create(Competition dto, String ownerId) throws DatastoreException {
 		CompetitionUtils.ensureNotNull(dto, "Competition object");
 		CompetitionUtils.ensureNotNull(ownerId, "Owner ID");
 		
-		// Convert to DBO
+		// convert to DBO
 		CompetitionDBO dbo = new CompetitionDBO();
 		copyDtoToDbo(dto, dbo);
 		
-		// Ensure name is not taken
-		if (lookupByName(dto.getName()) != null)
-			throw new IllegalArgumentException("Sorry, a Competition already exists with the name " + dto.getName());
-		
-		// Generate ID
+		// generate unique ID
 		dbo.setId(idGenerator.generateNewId());
 		
-		// TODO: Generate eTag
-		dbo.seteTag(eTagGenerator.generateETag(dbo));
-		
 		// set Owner ID
 		dbo.setOwnerId(Long.parseLong(ownerId));
 		
-		// Set creation date
+		// set creation date
 		dbo.setCreatedOn(System.currentTimeMillis());
 		
-		// Ensure DBO has required information
+		// generate eTag
+		tagMessenger.generateEtagAndSendMessage(dbo, ChangeType.CREATE);
+		
+		// ensure DBO has required information
 		verifyCompetitionDBO(dbo);
 		
-		// Create DBO
+		// create DBO
 		try {
 			dbo = basicDao.createNew(dbo);
 			return dbo.getId().toString();
 		} catch (Exception e) {
-			throw new DatastoreException("id="+dbo.getId()+" name="+dto.getName(), e);
+			String message = "id=" + dbo.getId() + " name=" + dto.getName();
+			
+			// check if a name conflict occurred
+			if (e.getClass() == IllegalArgumentException.class) {
```

The basic dao should throw the duplicate exception, not wrap it in an illegal arg.